src/mlpack/methods/neat/acyclic_net.hpp

```diff
@@ -0,0 +1,74 @@
+/**
+ * @file acyclic_net.hpp
+ * @author Rahul Ganesh Prabhu
+ *
+ * Definition of the Acyclic net classes.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_METHODS_NEAT_ACYCLIC_NET_HPP
+#define MLPACK_METHODS_NEAT_ACYCLIC_NET_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack{
+namespace neat /** NeuroEvolution of Augmenting Topologies */{
+
+template <class ActivationFunction>
+class AcyclicNet
+{
+ public:
+  /**
+   * Creates an AcyclicNet object.
+   * 
+   * @param NodeGeneList A vector storing all the node IDs.
+   * @param DirectedGraph A map of maps storing connection genes, whose first
+   *    key is the source ID, and second key is the target ID.
+   * @param actFn The activation function.
+   * @param inputNodeCount The number of input nodes.
+   * @param outputNodeCount The number of output nodes.
+   * @param bias The bias.s
+   */
+  AcyclicNet(std::vector<int>& NodeGeneList,
```

Comment:
I'm wondering if it would be more efficient to update the network instead of creating a new one for each evalution. let me know what you think.

Reply:
I agree, but I was not able to think of an appropriate way to do this during crossovers, since the entire network structure undergoes a change, unlike mutation where I could update depths for a single node or connection. I'll think about it some more though.

Reply:
What we could do is to update the overall network once we do crossovers and update the it in the other case. But ideally we can do both without rebuilding the network, I'll think about it as well.

Reply:
The thing is, mutation and crossover occur together during the reproduction step, so under the current system, we don't have to update for mutation either, since we aren't for crossover.

I can however remove the node gene list, since nodes can't be deleted, i just need to store the max node ID. I'm also working on the cyclic network evaluation right now.

Reply:
You are right thanks for the clarification.